File: backend/dtos/game_summary.go
package dtos

// GameSummary is the summary of a game. Has just enough information to render
// the game card.
//
// easyjson:json
type GameSummary struct {
	ID               string         `json:"1"`
	LiveGameStats    LiveGameStats  `json:"2"`
	GameStartTime    int            `json:"3"`
	GameStartTimeTBD bool           `json:"4"`
	Finished         bool           `json:"5"`
	NotStarted       bool           `json:"6"`
	HomeTeamStatus   GameTeamStatus `json:"7"`
	AwayTeamStatus   GameTeamStatus `json:"8"`
}

File: frontend/client/services/dtos/game-summary.js

import LiveGameStats from './live-game-stats'
import GameTeamStatus from './game-team-status'
import { Num, Bool, Text, DTO } from 'services/util/types'

export default DTO({
  id:               Text('1'),
  liveGameStats:    LiveGameStats('2'),
  gameStartTime:    Num('3'),
  gameStartTimeTBD: Bool('4'),
  finished:         Bool('5'),
  notStarted:       Bool('6'),
  homeTeamStatus:   GameTeamStatus('7'),
  awayTeamStatus:   GameTeamStatus('8'),
})
